lib/java/src/org/apache/thrift/TBaseAsyncProcessor.java

```diff
@@ -89,7 +89,7 @@ public boolean process(final AsyncFrameBuffer fb) throws TException {
               out.getTransport().flush();
             }
             fb.responseReady();
-            return true;
+            return;
         }
         in.readMessageEnd();
 
```

Comment:
Check out line 97 below - it looks like responseReady is called before we start processing the response.  Does that lead to optimal behavior or will the async server end up blocking trying to read the reply that doesn't exist?